### ISSUE TITLE
[java] Allow removing of break statement in last case of a switch statement

### ISSUE THREAD

**Issue by mmoehring** [pmd/pmd#797]
**Rule Set:**
Design: MissingBreakInSwitch

**Description:**
The Rule MissingBreakInSwitch will report an error if the last case in a switch statement has no break statement. The break statement is not needed in the last case of a switch.
It would be nice to have a parameter to allow this behavior.

**Code Sample demonstrating the issue:**
The default case should not trigger an error in the following code when the new parameter is set.

```java
switch(foo) {
  case OK:
    bar = "whatever";
    break;
  default:
    bar = "last case";
}
```

**Running PMD through:** *Gradle*
**PMD version:** *5.8.1*

**Comment by oowekyala** [comment]
This would indeed be a nice enhancement! Feel free to tackle it if you've got the time

**Comment by oowekyala** [comment]
Related: #659

**Comment by jsotuyod** [comment]
Closing this as duplicate in favor of #659

**Comment by Midya-ELTE** [comment]
It want to design a rule which is able to detect fall-through cases , any one can help me in that? -- please write for me midyaazad3@gmail.com

**Comment by oowekyala** [comment]
@Midya-ELTE #2894 already tracks this, and it's already been implemented, though not merged yet
